script/lib/env_util.py

```diff
@@ -58,6 +58,29 @@ def get_environment_from_batch_command(env_cmd, initial=None):
   proc.communicate()
   return result
 
+def get_vs_location(vs_version):
+    """
+    Returns the location of the VS building environment.
+
+    The vs_version can be strings like "[15.0,16.0)", meaning 2017, but not the next version.
+    The arch has to be one of "x86", "amd64", "arm", "x86_amd64", "x86_arm", "amd64_x86",
+    "amd64_arm", e.g. the args passed to vcvarsall.bat.
+    """
+
+    # vswhere can't handle spaces, like "[15.0, 16.0)" should become "[15.0,16.0)"
+    vs_version = vs_version.replace(" ", "")
+
+    # Find visual studio
+    proc = subprocess.Popen(
+      "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\vswhere.exe "
```

Comment:
This should preferably use the `%ProgramFiles(x86)%` env. variable instead of a hard-coded path. Even better the full install path would be read from the registry.

Reply:
You're right about using `%ProgramFiles(x86)%`.

From this documentation, it seems like we might want to prefer that solution over using the registry, too:

https://github.com/Microsoft/vswhere/wiki/Installing

>  Starting with Visual Studio 15.2 (26418.1 Preview) vswhere.exe is installed in %ProgramFiles(x86)%\Microsoft Visual Studio\Installer (use %ProgramFiles% in a 32-bit program prior to Windows 10). This is a fixed location that will be maintained.